enhancements/cluster-scope-secret-volumes/shared-resource-validation.md

```diff
@@ -0,0 +1,407 @@
+
+Any handling of existing Shared Resources whose names start with 'openshift-*' and were created when OCP was at a 
+4.10.x level will be done by the controller, and not the new webhooks.  The controller's existing watches, state around
+running Pods with Shared Resource volume mounts, is better suited to any actions of this ilk.  The controller will use the 
+new "validity" condition, with the name restriction as the reason, if an unapproved use of "openshift-*" is found,
+and will handle mounts and existing data according to the defaults for the 2 new boolean fields.  But let's discuss 
+if we should toggle that behavior.
+
+The webhook will handle the "openshift-*" naming restriction for creates of Shared Resources.
+The webhook will read the "allow list" ConfigMaps on startup (they should not change except on upgrade) and so its validations 
+should not involve any RPC calls.
+
+Existing checks in the controller for the presence of "readOnly: true" check will be kept.  Analogous checks for 
+who creates "openshift-*" Shared Resources will also exist in the controller.  This covers the case when the webhook
+might be down.
+
+Per open questions, discovery of how restricting creation of namespaces/projects starting with "openshift-*" should be
+done.  It is feasible that we augment the ConfigMap based "allow list" with the same mechanism (RBAC based?  Controller based?
+Project endpoint in the aggregated API Server?).
+
+### Risks and Mitigations
+
+Since Shared Resources is Tech Preview 4.10, we can introduce validation controls that "break existing API usage" in later releases than 4.10.
+In particular, any users who were creating SharedSecrets of their own that start with "openshift-".  But we have to 
+have all such restrictions in place before moving from Tech Preview to GA.  The 4.10 blog post does give a warning
+that we might restrict use of Shared Resources names starting with "openshift-*"
+
+And again, the "readOnly: true" validation was already in effect in 4.10, but only enforced at the controller.
+
+That said, even with Tech Preview, release notes are warranted if we go down the path of invalidating of 
+Shared Resources that were "OK" in 4.10.  And user push back is at least conceivable on a case by case basis.
+
+## Design Details
+
+
+### Open Questions [optional]
+
+ 1. Is there a "Pod Spec'able" filter wrt configuring validating admission webhooks, so we catch all the subtypes of Pod?  Or do we have to list each k8s type that has Volumes in the spec?  Do not think there is a "pod spec'able" filter, but let's confirm.
+ 2. Existing k8s metrics cited below are sufficient for execution times for each request/response interaction when our new admission validations
+    are called, correct? If not, does it make sense to have new metrics be histograms around the admission validations execution times?
+ 3. Validating webhooks can define / maintain metrics and a metric service, right?
```

Comment:
Yes, from metrics perspective, a webhook it's just a Pod that exposes a metric endpoint.

Reply:
thanks for confirmation